php-frontend/src/main/java/org/sonar/php/symbols/FunctionSymbolData.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.php.symbols;
+
+import org.sonar.plugins.php.api.symbols.QualifiedName;
+import org.sonar.plugins.php.api.tree.declaration.ParameterTree;
+import org.sonar.plugins.php.api.visitors.LocationInFile;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+public class FunctionSymbolData {
+  private final LocationInFile location;
+  private final QualifiedName qualifiedName;
+  private final List<Parameter> parameters;
+  private final boolean hasReturn;
+
+  public FunctionSymbolData(LocationInFile location, QualifiedName qualifiedName, List<Parameter> parameters, boolean hasReturn) {
+    this.location = location;
+    this.qualifiedName = qualifiedName;
+    this.parameters = parameters;
+    this.hasReturn = hasReturn;
+  }
+
+  public LocationInFile location() {
+    return location;
+  }
+
+  public QualifiedName qualifiedName() {
+    return qualifiedName;
+  }
+
+  public List<Parameter> parameters() {
+    return parameters;
+  }
+
+  public boolean hasReturn() {
+    return hasReturn;
+  }
+
+  public static class Parameter {
+    private final String name;
+    private final String type;
```

Comment:
You're right. Since the types can be NamespaceTrees or BuiltInTypeTrees, I wasn't sure how to best combine them. I wanted to wait until we actually make use of the types, to potentially improve the design here.